### PR TITLE
Feauture/expanding buttons

### PR DESCRIPTION
**Pull request by MsweliP**
Prevent UI from unnecessary loads

### CODE REVIEW

File: waterlog/src/Components/Wastage.js [hunk 1]
```diff
@@ -55,11 +55,14 @@ class Wastage extends Component {
       seasonalLoading,
       placeholderLoading,
       placeholderError,
+      dailyWaste,
+      monthlyWaste,
+      seasonWaste,
```
Comment (GTaylorRabbit) [inline waterlog/src/Components/Wastage.js]: Please note the comma after seasonWaste as it is the last item

Comment (MsweliP) [reply waterlog/src/Components/Wastage.js]: removed , please re-check

**Review by GTaylorRabbit — APPROVED** [review]
I think if you made a note that these changes were for adding more defined checks for the expanding buttons, it would better describe the PR.

File: waterlog/src/Components/Usage.js [hunk 2]
```diff
@@ -87,16 +87,25 @@ class Usage extends Component {
...
+      monthlyUsage,
+      monthlyCost,
+      seasonUsage,
+      seasonalCost,
+      forecastDaily,
+      forecastMonthly,
       seasonalLoading } = this.props;
 
-    if (dailyError || monthlyError
+    if ((dailyError || monthlyError
```
Comment (FredJEhlers) [inline waterlog/src/Components/Usage.js]: Please make this more readable 
I have dm's you a possible fix

File: waterlog/src/Components/Usage.js [hunk 3]
```diff
@@ -87,16 +87,25 @@ class Usage extends Component {
...
       || seasonalError || dailyCostForecastError
-      || monthlyCostForecastError || placeholderError) {
-      return <div><Error404/></div>;
+      || monthlyCostForecastError || placeholderError)&& (dailyUsage.length < 1)&& (monthlyUsage.length < 1)&& (seasonUsage.length < 1)
+      && (dailyCost.length < 1)&& (monthlyCost.length < 1)&& (seasonalCost.length < 1)) {
+      return <div>Error!</div>;
     }
-    if (dailyLoading || monthlyLoading
+    if ((dailyLoading || monthlyLoading
```
Comment (FredJEhlers) [inline waterlog/src/Components/Usage.js]: same here

File: waterlog/src/Components/Wastage.js [hunk 4]
```diff
@@ -55,11 +55,14 @@ class Wastage extends Component {
...
       placeholderError,
+      dailyWaste,
+      monthlyWaste,
+      seasonWaste
     } = this.props;
     if (dailyError || monthlyError || seasonalError || placeholderError) {
       return <Error404 />;
     }
-    if (dailyLoading || monthlyLoading || seasonalLoading || placeholderLoading) {
+    if ((dailyLoading || monthlyLoading || seasonalLoading || placeholderLoading) && (dailyWaste.length < 1) && (monthlyWaste.length < 1)&& (seasonWaste.length < 1)) {
```
Comment (FredJEhlers) [inline waterlog/src/Components/Wastage.js]: again here as well

**Review by FredJEhlers — CHANGES_REQUESTED** [review]

File: waterlog/src/Components/Usage.js (hunk 3 above)
Comment (MsweliP) [reply waterlog/src/Components/Usage.js]: Done

File: waterlog/src/Components/Usage.js (hunk 2 above)
Comment (MsweliP) [reply waterlog/src/Components/Usage.js]: Done

File: waterlog/src/Components/Wastage.js (hunk 4 above)
Comment (MsweliP) [reply waterlog/src/Components/Wastage.js]: Done

Comment (MsweliP) [reply waterlog/src/Components/Wastage.js]: Done

**Review by FredJEhlers — APPROVED** [review]

**Pull request merged**